integration/test_batch.py

```diff
@@ -0,0 +1,120 @@
+import uuid
+
+import pytest
+
+import weaviate
+
+
+@pytest.fixture(scope="module")
+def client():
+    client = weaviate.Client("http://localhost:8080")
+    yield client
+    client.schema.delete_all()
+
+
+def test_add_data_object(client: weaviate.Client):
```

Comment:
I'd propose to do the test differently (same for references)

You can parameterize the input like this:
```
@pytest.mark.parametrize("vector", [None, [1,2,3]])
@pytest.mark.parametrize("uuid", [None, 1, 5, 20, 50])
def test_add_data_object(client: weaviate.Client, uuid: TYPE, vector: TYPE):
    client.batch.add_data_object(
        data_object={},
        class_name="Test",
       uuid= uuid
       vector=vector
    )
```

and then run the test for all of these combinations automatically.

I'd also recommend to move the testing of the number of objects to a sparate test. Then it is possible to test each functionality on its own and also run them on its own

Reply:
This looks better, I will look into it.